File: MyRixi-API/Models/Message.cs
namespace MyRixiApi.Models;

public class Message
{
    public Guid Id { get; set; }
    public string Content { get; set; } = string.Empty;
    public DateTime SentAt { get; set; } = DateTime.UtcNow;
    public bool IsRead { get; set; } = false;
    
    // Remplacer la référence à Conversation par Channel
    public Guid ChannelId { get; set; }
    public Channel Channel { get; set; } = null!;
    
    public Guid SenderId { get; set; }
    public User Sender { get; set; } = null!;
    
    // Ajoutons la possibilité de joindre des médias
    public ICollection<Media> Attachments { get; set; } = new List<Media>();
    
    // Réactions aux messages
    public ICollection<MessageReaction> Reactions { get; set; } = new List<MessageReaction>();
}

File: MyRixi-Mobile/types/message.ts
import { Channel } from "./channel";
import { Media } from "./media";
import { UserChannel } from "./profile";
import { Reaction } from "./reaction";

export interface Message {
  id: string;
  content: string;
  sentAt: Date;
  isRead: boolean;
  channelId: string;
  channel?: Channel;
  sender?: UserChannel;
  attachments: Media[];
  reactions: Reaction[];
  isPending?: boolean;
}

export interface CreateMessageDto {
  content: string;
  channelId: string;
  attachmentIds: string[];
}